modules/circe/src/main/scala/evo/derivation/circe/deriveEnumerationCodec.scala

```diff
@@ -0,0 +1,79 @@
+package evo.derivation.circe
+
+import evo.derivation.LazySummon
+import evo.derivation.config.Config
+import evo.derivation.internal.underiveableError
+import evo.derivation.internal.showType
+import evo.derivation.internal.Matching
+import io.circe.*
+
+import scala.compiletime.*
+import scala.compiletime.error
+import scala.deriving.Mirror
+
+private inline def underiveableEnumerationCodec[A]: Nothing =
+    error("could not derive " + showType[Codec[A]] + ", " + showType[A] + "'s, children cannot have fields.")
+
+inline def deriveEnumerationCodec[A]: Codec[A] = EnumerationCodec.derived[A]
+
+private trait EnumerationCodec[A] extends Codec[A]
+private object EnumerationCodec:
+    inline def derived[A]: EnumerationCodec[A] =
+        given config: Config[A] = Config.derived[A]
+        summonFrom {
+            case given Mirror.ProductOf[A] =>
+                deriveForProduct[A]
+            case given Mirror.SumOf[A]     =>
+                // def hasFields = config.constructors.values.exists(_.fieldNames.nonEmpty)
+                // if hasFields then underiveableEnumerationCodec[A]
+                // else deriveForSum[A]
+                deriveForSum[A]
```

Comment:
how to prevent from deriving enumeration codec if children have some fields? If I uncomment that if, then compilation fails even for valid cases

Reply:
There is one trick. You can require ValueOf for each of the subtypes to prove that enum is simple
https://scastie.scala-lang.org/Odomontois/tjXaZnPoRZK7pRcPpedGyw

Reply:
I guess general enumeration stuff is overall useful, so we can move this detection to the core Config

Reply:
Ok done in #39